File: 3.java
import java.util.HashSet;

class Solution {
    public int lengthOfLongestSubstring(String s) {
        if(s.length() == 0) {
            return 0;
        }
        int max = 1;
        for(int i = 0; i < s.length(); i++) {
            HashSet<Character> letters = new HashSet<>();
            letters.add(s.charAt(i));
            for(int j = i + 1; j < s.length(); j++) {
                if(letters.contains(s.charAt(j))) {
                    break;
                }
                letters.add(s.charAt(j));
            }
            if(max < letters.size()) {
                max = letters.size();
            }
        }
        return max;
    }
}

File: Python/3.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        map = {}
        left = 0
        result = 0
        for right in range(len(s)):
            if s[right] in map:
                left = max(map[s[right]] + 1, left)
            map[s[right]] = right
            result = max(result, right - left + 1)

        return result
